Showcase/Application/Foundations/Assets/AssetsViewController.swift

```diff
@@ -0,0 +1,465 @@
+        layout.minimumInteritemSpacing = 1
+        layout.minimumLineSpacing = 1
+        layout.estimatedItemSize = .zero
+        self.init(collectionViewLayout: layout)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        sections = makeSections()
+
+        view.backgroundColor = VitaminColor.Core.Background.primary
+
+        navigationItem.title = "Assets"
+        collectionView.register(
+            UINib(
+                nibName: "AssetCollectionViewCell",
+                bundle: nil),
+            forCellWithReuseIdentifier: "asset")
+        collectionView.register(
+            UINib(
+                nibName: "LargeAssetCollectionViewCell",
+                bundle: nil),
+            forCellWithReuseIdentifier: "largeAsset")
+
+        collectionView.register(
+            UICollectionReusableView.self,
+            forSupplementaryViewOfKind: UICollectionView.elementKindSectionHeader,
+            withReuseIdentifier: "header")
+
+        if let flow = self.collectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            flow.headerReferenceSize = CGSize(width: view.frame.width, height: 55)
+        }
+    }
+}
+
+// MARK: - UICollectionViewDelegateFlowLayout
+extension AssetsViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        sizeForItemAt indexPath: IndexPath
+    ) -> CGSize {
+        let defaultSize = CGSize(width: view.frame.width / 5, height: 90)
+        guard
+            let assetItem = sections[indexPath.section].items[indexPath.row] as? AssetItem
+        else {
+            return defaultSize
+        }
+        switch assetItem.type {
+        case .large:
+            return CGSize(width: view.frame.width, height: 110)
+        case.medium:
+            return CGSize(width: view.frame.width / 2.1, height: 90)
+        case .standard:
+            return defaultSize
+        }
+    }
+
+    func collectionView(
+        _ collectionView: UICollectionView,
+        layout collectionViewLayout: UICollectionViewLayout,
+        insetForSectionAt section: Int
+    ) -> UIEdgeInsets {
+        UIEdgeInsets(top: 0, left: 8, bottom: 0, right: 8)
+    }
+}
+
+// MARK: - UICollectionViewDataSource
+extension AssetsViewController {
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        sections.count
+    }
+
+    override func collectionView(
+        _ collectionView: UICollectionView,
+        numberOfItemsInSection section: Int
+    ) -> Int {
+        sections[section].items.count
+    }
+
+    override func collectionView(
+        _ collectionView: UICollectionView,
+        cellForItemAt indexPath: IndexPath
+    ) -> UICollectionViewCell {
+        guard let assetItem = sections[indexPath.section].items[indexPath.row] as? AssetItem else {
+            return AssetCollectionViewCell()
+        }
```

Comment:
Maybe use a `fatalErorr` if there is no `AssetItem`.

Reply:
OK, even if that should never occur, that's just a code written to avoid force unwrap ;)
Since this minor modification would erase your approval, I'll do that in a next PR